File: Corte 1/reto6.py
# Reto 6 - Calcular el total de una compra de varios productos

def main():
    n = int(input("Cuantos productos va a comprar? "))
    total = 0.0   # acumulador del total de la compra

    for i in range(n):
        print(f"--- Producto {i + 1} ---")
        nombre = input("Nombre del producto: ")
        precio = float(input("Precio unitario: "))
        cantidad = int(input("Cantidad comprada: "))

        subtotal = precio * cantidad
        print(f"Subtotal de {nombre}: {subtotal}")

        total += subtotal   # acumula

    # Descuento segun el total
    if total > 300000:
        descuento = 0.10
    elif total >= 150000:          # aqui ya sabemos que total <= 300000
        descuento = 0.05
    else:
        descuento = 0.0

    valorDescuento = total * descuento
    totalPagar = total - valorDescuento

    print("===== RESUMEN =====")
    print(f"Total antes del descuento: {total}")
    print(f"Descuento aplicado: {descuento * 100:.0f}% ({valorDescuento})")
    print(f"Total a pagar: {totalPagar}")


if __name__ == "__main__":
    main()

File: Corte 1/Retos/reto6.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

int main() {
    int n;
    cout << "Cuantos productos va a comprar? ";
    cin >> n;
    cin.ignore();   // limpia el salto de linea antes de leer con getline

    double total = 0.0;   // acumulador

    for (int i = 0; i < n; i++) {
        cout << "--- Producto " << (i + 1) << " ---" << endl;

        string nombre;
        double precio;
        int cantidad;

        cout << "Nombre del producto: "; getline(cin, nombre);
        cout << "Precio unitario: "; cin >> precio;
        cout << "Cantidad comprada: "; cin >> cantidad;
        cin.ignore();   // limpia el salto antes del siguiente getline

        double subtotal = precio * cantidad;
        cout << "Subtotal de " << nombre << ": " << subtotal << endl;

        total += subtotal;   // acumula
    }

    // Descuento segun el total
    double descuento;
    if (total > 300000) descuento = 0.10;
    else if (total >= 150000) descuento = 0.05;   // total <= 300000 en esta rama
    else descuento = 0.0;

    double valorDescuento = total * descuento;
    double totalPagar = total - valorDescuento;

    cout << fixed << setprecision(2);
    cout << "===== RESUMEN =====" << endl;
    cout << "Total antes del descuento: " << total << endl;
    cout << "Descuento aplicado: " << descuento * 100 << "% (" << valorDescuento << ")" << endl;
    cout << "Total a pagar: " << totalPagar << endl;

    return 0;
}